elastic/beats
# Expose filestream metrics together with log input

## What does this PR do?

The filestream input now also exposes the following metrics:
- filebeat.harvester.closed
- filebeat.harvester.open_file
- filebeat.harvester.started
- filebeat.harvester.running


## Why is it important?

The filestream input did not expose those metrics, which is very confusing, especially for users migrating from the log input

## Checklist

~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~

~~## Author's Checklist~~
## How to test this PR locally
1. Start Filebeat and ensure the HTTP endpoint is enabled, here is a config example:
```yaml
filebeat.inputs:
  - type: filestream
    id: my-filestream-id
    enabled: true
    close.reader.after_interval: 30s
    paths:
      - /tmp/filestream*.log

output.console:
  codec.json:
    pretty: true

logging:
  to_stderr: true

http:
  enabled: true
  host: localhost
  port: 5066
```

2. Create files matching the glob `/tmp/filestream*.log`.
3. Check the metrics with `curl -sL http://localhost:5066/stats | jq '.[].harvester | select(.!=null)'`
4. Check the monitoring log metrics contains the correct values, look for entries like:
```
{"log.level":"info","@timestamp":"2023-06-20T18:36:14.044+0200","log.logger":"monitoring","log.origin":{"file.name":"log/log.go","file.line":187},"message":"Non-zero metrics in the last 30s","service.name":"filebeat","monitoring":{"metrics":{"beat":{"cgroup":{"cpu":{"id":"emacs.service"},"memory":{"id":"emacs.service","mem":{"usage":{"bytes":26226442240}}}},"cpu":{"system":{"ticks":20,"time":{"ms":20}},"total":{"ticks":30,"time":{"ms":30},"value":30},"user":{"ticks":10,"time":{"ms":10}}},"handles":{"limit":{"hard":524288,"soft":524288},"open":12},"info":{"ephemeral_id":"402cd7c0-145a-48fd-bda2-553f979af7a3","name":"filebeat","uptime":{"ms":30016},"version":"8.9.0"},"memstats":{"gc_next":11980576,"memory_alloc":7224144,"memory_sys":26326280,"memory_total":15859808,"rss":64663552},"runtime":{"goroutines":32}},"filebeat":{"events":{"active":0,"added":2,"done":2},"harvester":{"open_files":2,"running":2,"started":2}},"libbeat":{"config":{"module":{"running":0}},"output":{"events":{"acked":1,"active":0,"batches":1,"total":1},"type":"console","write":{"bytes":616}},"pipeline":{"clients":3,"events":{"active":0,"filtered":1,"published":1,"total":2},"queue":{"acked":1,"max_events":4096}}},"registrar":{"states":{"current":1,"update":1},"writes":{"success":1,"total":1}},"system":{"cpu":{"cores":16},"load":{"1":4.3,"15":3.85,"5":4.06,"norm":{"1":0.2688,"15":0.2406,"5":0.2537}}}},"ecs.version":"1.6.0"}}
```


## Related issues
- Closes #33771

~~## Use cases~~
~~## Screenshots~~
~~## Logs~~
